coders/xcf.c

```diff
@@ -1086,11 +1157,18 @@ static Image *ReadXCFImage(const ImageInfo *image_info,ExceptionInfo *exception)
       (LocaleNCompare((char *) magick,"gimp xcf",8) != 0))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   (void) memset(&doc_info,0,sizeof(XCFDocInfo));
+  doc_info.version=strtoul(magick+10, NULL, 10);
```

Comment:
Could you change the size of `magick` to `10` and use `sizeof(magick)` when reading it instead of the current value `14`. And then use `ReadBlobMSBLong(image)` here?